tests/unit/test_client.py

```diff
@@ -864,6 +864,29 @@ def test_extra_credential_value_encoding(mock_get_and_post):
     assert headers[constants.HEADER_EXTRA_CREDENTIAL] == "foo=bar+%E7%9A%84"
 
 
+def test_extra_credential_value_object(mock_get_and_post):
+    _, post = mock_get_and_post
+
+    class TestCredential(object):
+        def __str__(self):
+            return "bar"
+
+    req = TrinoRequest(
+        host="coordinator",
+        port=constants.DEFAULT_TLS_PORT,
+        client_session=ClientSession(
+            user="test",
+            extra_credential=[("foo", TestCredential())]
```

Comment:
I get why you made this change, makes it easier to use object as cred values. But how is this different from below? Just want to understand what problem it actually solves.

```
extra_credential=[("foo"), str(TestCredential())]
```

Reply:
If `TestCredential()` provides a time limited credential (JWT or otherwise) then the object can refresh the underlying credential. This has two benefits:

1. No need to regenerate TrinoClient object.
1. If the time-limited credential expiry is hit, the automatic retry should include the refreshed credential

Reply:
Somehow I missed that your comment was on the test! I've added a second assertion which hopefully is both a more thorough test and shows the difference clearly.